plugins/ida/ida_main.ml

```diff
@@ -136,13 +136,103 @@ let loader path =
             else code, Memmap.add data mem sec) in
   Project.Input.create arch path ~code ~data
 
+let int64_to_word arch = 
+  let addr_size = Arch.addr_size arch in 
+  (Word.of_int64 ~width:(Size.in_bits addr_size)) 
+
+(* copied from lib/bap_disasm/bap_disasm_brancher.ml *)
+let kind_of_dests = function 
+  | xs when List.for_all xs ~f:(fun (_,x) -> x = `Fall) -> `Fall 
+  | xs -> if List.exists  xs ~f:(fun (_,x) -> x = `Jump) 
+    then `Jump 
+    else `Cond 
+
+let print_addr_with_dests addr dests = 
+  Format.printf "%x: %a\n" 
+    (Int64.to_int_exn addr) 
+    Sexp.pp (Brancher.sexp_of_dests dests) 
+
+(* brancher_info is a list of 3-tuples of the form 
+ * (addr, jump of normal flow (fall) type, jumps of other types) *) 
+type brancher_info = (int64 * int64 option * int64 list) list 
+[@@deriving sexp] 
+
+module BrancherInfo = Data.Make(struct 
+    type t = brancher_info 
+    let version = "0.1" 
+  end) 
+
+let read_brancher_info name = 
+  In_channel.with_file name ~f:(fun ch -> 
+      Sexp.input_sexp ch |> brancher_info_of_sexp) 
+
+let load_brancher_info = 
+  Command.create 
+    `python 
+    ~script:(request "brancher") 
+    ~parser:read_brancher_info 
+
+let handle_normal_flow flow arch = 
+  let (!) = (int64_to_word arch) in 
+  match flow with 
+  | Some fall -> [Some !fall, `Fall] 
+  | None -> [] 
+
+let handle_other_flows flows default_dests arch = 
+  let (!) = (int64_to_word arch) in 
+  let helper acc dest_addr = 
+    let default_edge = ListLabels.assoc_opt 
+        (Some !dest_addr) 
+        default_dests 
+    in 
+    match default_edge with 
+    (* If dest_addr is also found by the default BIL brancher, 
+     * we reuse the edge type *) 
+    | Some kind -> (Some !dest_addr, kind)::acc 
+    (* Else we use a heuristic to guess the edge type *) 
+    | None -> (Some !dest_addr, (kind_of_dests default_dests))::acc 
```

Comment:
Note that I use a different heuristic from #550, which assumed that the unknown type was a jump.

Reply:
I don't think that we need any heuristic here, as soon as we have a `Fall` edge, it shouldn't really matter whether a jump is conditional or not, or do you experience any problems when you've set all to `Jump` kind?

If indeed it has negative impact, then you can use `Insn.(is_conditional insn)` and mark all edges with other flow as `Cond` if it is true. (Note, this predicate will use instruction semantics (i.e., BIL) if available, and will fallback to the semantic information available from the disassembler backend).

Reply:
Setting all to jump works, but its just a little weird that the Brancher module expects more detailed information i.e. jump types but does not use it.